### PR TITLE
Update endpoint slices docs

### PR DESCRIPTION
**Pull request by aojea**
Be more explicit on the "duplicate endpoints" sections about what are the consumers expectations and the need to parse all the list of slices and remove only the duplicates

/sig network
/assign @robscott @thockin @sftim

### CODE REVIEW

File: content/en/docs/concepts/services-networking/endpoint-slices.md [hunk 1]
```diff
@@ -233,11 +233,12 @@ different EndpointSlice objects can arrive at the Kubernetes client watch / cach
 at different times.
 
 {{< note >}}
-Clients of the EndpointSlice API must be able to handle the situation where
-a particular endpoint address appears in more than one slice.
+Clients of the EndpointSlice API must iterate through all the existing EndpointSlices
+associated to a Service and build a complete list of unique network endpoints. It is
```
Comment (shannonxtreme) [inline content/en/docs/concepts/services-networking/endpoint-slices.md:237]: nit

```suggestion
associated with a Service and build a complete list of unique network endpoints. It is
```

File: content/en/docs/concepts/services-networking/endpoint-slices.md [hunk 2]
```diff
@@ -233,11 +233,12 @@ different EndpointSlice objects can arrive at the Kubernetes client watch / cach
...
+important to mention that endpoints may be duplicated in different EndointSlices.
```
Comment (shannonxtreme) [inline content/en/docs/concepts/services-networking/endpoint-slices.md:238]:
```suggestion
associated to a Service and build a complete list of unique network endpoints. Endpoints might be duplicated in different EndointSlices.
```

**Review by shannonxtreme — COMMENTED** [review]
One nit but the other will improve readability I think

**Comment by shannonxtreme** [comment]
Binding LGTM otherwise 😁

File: content/en/docs/concepts/services-networking/endpoint-slices.md (hunk 2 above)
Comment (shannonxtreme) [reply content/en/docs/concepts/services-networking/endpoint-slices.md:238]: Done ☺️

**Comment by robscott** [comment]
Thanks @aojea!

/lgtm

**Review by thockin — COMMENTED** [review]
/lgtm

**Pull request merged**